### ISSUE TITLE
language detect problem

### ISSUE THREAD

**Issue by chenjeff622** [Ylianst/MeshAgent#156]
we use the last meshcentral agent. it always display english UI when start agent. we open MSH file and it contains our language code(zh-hant).  we expect agent will start with our language(zh-hant) interface. so how do i debug the problem?

**Comment by krayon007** [comment]
Which OS are you running this on?

**Comment by chenjeff622** [comment]
windows 2008R2

**Comment by krayon007** [comment]
What does the OS report, if you run the following command from the command prompt:

`wmic os get oslanguage /FORMAT:LIST`

**Comment by chenjeff622** [comment]
OSLanguage=1028

**Comment by chenjeff622** [comment]
> What does the OS report, if you run the following command from the command prompt:
> 
> `wmic os get oslanguage /FORMAT:LIST`

OSLanguage=1028

**Comment by krayon007** [comment]
1028 is zh-TW. 
zh-hant is 31748

Is there an entry for zh-TW in your MSH file?

**Comment by chenjeff622** [comment]
No, but zh-hant font code is correct. How do i add zh-TW font code copy from zh-hant?

**Comment by krayon007** [comment]
[This video](https://www.youtube.com/watch?v=SqQEI6v84ZU) that Ylian posted shows how to update the language translations. I'll see if I can some other documentation that shows more stuff more specific to where the agent translations are on the server.

**Comment by chenjeff622** [comment]
> [This video](https://www.youtube.com/watch?v=SqQEI6v84ZU) that Ylian posted shows how to update the language translations. I'll see if I can some other documentation that shows more stuff more specific to where the agent translations are on the server.

it seem translation web language only. i try to add zh-TW json string into meshagent64-test.msh . it shows english still when start meshagent64-test.exe .  anyone can help me ?

**Comment by si458** [comment]
this issue was discussed in meshcentral repo (will edit when i find issue)
basically if you installed your original windows with 1 language, but then switched languages via language packs,
it would still load the old language in the ui because we read the wrong value
